### PR TITLE
 #15685 made peer_test.go more portable by using random free port instead of hardcoded port 30303

### PR DESCRIPTION
**Pull request by original-brownbear**
Improves test portability for #15685 by resolving `127.0.0.1:0` to get a random free port instead of the hard coded one => now the test works if you have a running node on the same interface already.

### CODE REVIEW

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 82e2a4aa3354213eaf7956b6b1c915e138de42ca
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thank you for your contribution! Your commits seem to not adhere to the repository coding standards

 * Commit: 8675312c5ec1734b854863815c3b17b178c143a6
  * Your commit message body contains a line that is longer than 80 characters
  * Commits must be prefixed with the package(s) they modify

Please check the [contribution guidelines](https://github.com/ethereum/go-ethereum#contribution) for more details.

---
This message was auto-generated by https://gitcop.com

File: whisper/whisperv5/peer_test.go [hunk 1]
```diff
@@ -108,8 +108,7 @@ func TestSimulation(t *testing.T) {
 
 func initialize(t *testing.T) {
 	var err error
-	ip := net.IPv4(127, 0, 0, 1)
-	port0 := 30303
+	ip, port0 := getFreeAddress(t)
```
Comment (fjl) [inline whisper/whisperv5/peer_test.go]: `p2p.Server` sets `Port` to the actual port after startup. Please use `Port: 0` in config, then call `AddPeer` after they're all started.

**Comment by original-brownbear** [comment]
@fjl thanks for the review, fixed I think :)

**Comment by original-brownbear** [comment]
@fjl ping :)

File: whisper/whisperv5/peer_test.go [hunk 2]
```diff
@@ -159,6 +142,15 @@ func initialize(t *testing.T) {
 			t.Fatalf("failed to start server %d.", i)
 		}
 
+		if i > 0 {
+			peerNodeId := nodes[i-1].id
+			address, _ := net.ResolveTCPAddr("tcp", nodes[i-1].server.ListenAddr)
+			peerPort := uint16(address.Port)
+			peerNode := discover.PubkeyID(&peerNodeId.PublicKey)
+			peer := discover.NewNode(peerNode, address.IP, peerPort, peerPort)
+			node.server.AddPeer(peer)
```
Comment (gballet) [inline whisper/whisperv5/peer_test.go]: You're changing the behavior of the tests here: each node has only one peer (the one coming before) while bootstrap + static + trusted nodes were all previous nodes.

**Comment by original-brownbear** [comment]
@gballet np, thanks for the review! Will fix this up tomorrow :)

**Comment by original-brownbear** [comment]
@gballet fixed (I hope :)):

* made it connect to all previously created peers again
* Adjusted the v6 tests to work the same way
   * Had to make them not start the servers in parallel to allow for that, hope that's ok

**Comment by gballet** [comment]
@original-brownbear could you clarify why you had to remove the goroutines and `waitForServersToStart`?

**Comment by original-brownbear** [comment]
@gballet 

because I need to have the servers actually up and running for them to resolve the wildcard `0` port to a real port that I can connect to, don't I?

**Comment by gballet** [comment]
I see. What you could do is to keep servers in a goroutine and wait for them to start before you continue the loop. I seem to remember that the intent behind this approach was to see if we caught some non-deterministic issues.

**Comment by gballet** [comment]
@fjl could you please check if you still have issues with this PR, so that we can merge it?

**Comment by original-brownbear** [comment]
@gballet thanks + sorry for not getting back to this earlier!

**Pull request merged**